pkg/network/ebpf/c/http.h

```diff
@@ -108,6 +108,7 @@ static __always_inline int http_begin_request(http_transaction_t *http, http_met
     // This can happen in the context of HTTP keep-alives;
     if (http_responding(http)) {
         http_enqueue(http, tup);
+        bpf_map_update_elem(&http_in_flight, tup, http, BPF_NOEXIST);
```

Comment:
Can you explain why this is needed? The last line of http_enqueue deletes from this same map.

Reply:
So when you have multiple requests over the same socket the following was happening:
1) Map entry is created with _request_ 1 information;
2) Map entry is updated with _response_ 1 information (but not flushed because there is no FIN in the TCP segment)
Then when we start to process request 2:
1) we retrieve the `http` object and `http_responding(http)` evaluates to true, so we flush transaction 1 and delete the map entry;
2) Now we populate the fields of `http` with _request_ 2 information, but the `http` object no longer points to a map value reference, so this information is lost.

That's why we need this map update

Reply:
argh, I'll have to restructure the code because of Kernel 4.4:
```
R3 type=map_value expected=fp
```
Newer kernels seem to be fine with "re-adding" the map entry, but Kernel 4.4 requires the value being added to point to frame pointer.
I think memcpy will likely be problematic, so I'll try to move the `delete` out of `http_enqueue`